components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -352,6 +376,8 @@ class GeckoEngineSession(
 
             currentUrl = url
             initialLoad = false
+            pendingLoadRequest = null
```

Comment:
`onLocationChange` seems to work best here. `onPageStop` (with success==true) would be another option but that didn't work reliably for me when testing.